### PR TITLE
fix: 235 layers not calculated correctly waterside

### PR DESCRIPTION
**Pull request by Carsopre**
## Issue addressed
Solves #235 

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the KOSWAT team.

## What has been done?
- When a new geometry upper points (representing a "coating layer surface") intersects its previous geometry then we calculate the added geometry based on the intersection with the very first point that discerns from its base relative.

### Checklist
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
...

### CODE REVIEW

**Review by ArdtK — CHANGES_REQUESTED** [review]
Few comments.
In addition some general remarks:
- Would you add a legend to explain the colored dots and lines on the profile plots?
- A README on subpackage `reinforcement_layers` would be useful, with an explanation on what e.g. a core, layer, relative geometry, etc are.

File: .gitignore
Comment (ArdtK) [inline .gitignore:1]: Does this folder exist?

File: koswat/dike_reinforcements/reinforcement_layers/standard_reinforcement_layers_wrapper_builder.py
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/dike_reinforcements/reinforcement_layers/standard_reinforcement_layers_wrapper_builder.py:1]: Although my time is up I'll do it.

**Comment by Carsopre** [comment]
> Few comments. In addition some general remarks:
> 
>     * Would you add a legend to explain the colored dots and lines on the profile plots?
> 
>     * A README on subpackage `reinforcement_layers` would be useful, with an explanation on what e.g. a core, layer, relative geometry, etc are.

Unfortunately I do not have time for the legend nor the readme, these will be addressed in a separate issue as already discussed earlier this week: #239

**Pull request merged**